### ISSUE TITLE
[BUG] systemd unit checks failing; old godbus dependency?

### ISSUE THREAD

**Issue by mnot** [DataDog/datadog-agent#16058]
**Agent Environment**

> Agent 7.43.1 - Commit: 9e9c790 - Serialization version: v5.0.67 - Go version: go1.19.6

>  \> uname -a
> Linux cloud.mnot.net 6.1.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 14 Feb 2023 22:08:08 +0000 x86_64 GNU/Linux

**Describe what happened:**

Systemd unit checks fail; typical logs:

~~~
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:367 in submitBasicUnitMetrics) | Error getting unit unitProperties: http-cache-tests.service: dbus: wire format error: unexpected EOF
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:402 in submitPropertyMetricsAsGauge) | Error getting detailed properties for unit http-cache-tests.service
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:367 in submitBasicUnitMetrics) | Error getting unit unitProperties: prosody.service: dbus: connection closed by user
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:402 in submitPropertyMetricsAsGauge) | Error getting detailed properties for unit prosody.service
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:367 in submitBasicUnitMetrics) | Error getting unit unitProperties: httpd.service: dbus: connection closed by user
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:402 in submitPropertyMetricsAsGauge) | Error getting detailed properties for unit httpd.service
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:367 in submitBasicUnitMetrics) | Error getting unit unitProperties: postfix.service: dbus: connection closed by user
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:402 in submitPropertyMetricsAsGauge) | Error getting detailed properties for unit postfix.service
Mar 11 23:41:09 cloud.mnot.net agent[2696310]: 2023-03-11 23:41:09 UTC | CORE | WARN | (pkg/collector/corechecks/systemd/systemd.go:367 in submitBasicUnitMetrics) | Error getting unit unitProperties: redbot.service: dbus: connection closed by user
~~~

**Describe what you expected:**

Systemd unit checks to pass.

**Steps to reproduce the issue:**

Configure some systemd unit checks.

**Additional environment details (Operating System, Cloud provider, etc):**

Digging through bug reports on godbus, https://github.com/godbus/dbus/issues/85 seems relevant; it appears they fixed this around v5. I am not a go person, but if I read [go.mod](https://github.com/DataDog/datadog-agent/blob/main/go.mod) correctly, 4.10 is used:

> github.com/godbus/dbus v4.1.0+incompatible

Intriguingly, both 4.1.0 and 5.0.6 appear to be in my binary:

~~~
> go version -m `which datadog-agent` | grep godbus
	dep	github.com/godbus/dbus	v4.1.0+incompatible	h1:WqqLRTsQic3apZUK9qC5sGNfXthmPXzUZ7nQPrNITa4=
	dep	github.com/godbus/dbus/v5	v5.0.6	h1:mkgN1ofwASrYnJ5W6U/BxG15eXXXjirgZc7CLqkcaro=
~~~

However (and again, I'm not a go person), [systemd dbus_conn.go appears to refer](https://github.com/DataDog/datadog-agent/blob/76f8e77059fee758a8b03e0fc8dd4e879d6d5eb4/pkg/collector/corechecks/systemd/dbus_conn.go#L23) to the unqualified v4:

~~~
	godbus "github.com/godbus/dbus"
~~~

That's just my uneducated guess as to what's happening, might be completely wrong.

**Comment by Gigioliva** [comment]
Any news on this?

**Comment by DaveWK** [comment]
this is still broken in 7.47.0

**Comment by kivengh** [comment]
This is still broken in 7.49.1. I got this error on Amazon Linux 2023.

**Comment by iliakur** [comment]
ah, forgot to update this, but it should be fixed in the upcoming 7.52 release of the agent. I'll close the issue, plz reopen if you're still experiencing a problem **after an upgrade**.
